### PR TITLE
fix typos in DatabricksSubmitRunOperator

### PR DESCRIPTION
**Pull request by adam133**
Finally got around to trying this, realized I had a couple major typos in my original PR. Works to execute by pipeline name after these two changes.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by josh-fell — COMMENTED** [review]
Would you mind adding tests for the DatabricksHook to cover this for the future please?

File: airflow/providers/databricks/operators/databricks.py [hunk 1]
```diff
@@ -521,7 +521,7 @@ def execute(self, context: Context):
         ):
             # If pipeline_id is not provided, we need to fetch it from the pipeline_name
             pipeline_name = self.json["pipeline_task"]["pipeline_name"]
-            self.json["pipeline_task"]["pipeline_id"] = self._hook.get_pipeline_id(pipeline_name)
```
Comment (hussein-awala) [inline airflow/providers/databricks/operators/databricks.py:524]: I wonder why Mypy didn't block #32903 which added this call to the non-existing method. @potiuk do you have an idea?

**Review by hussein-awala — COMMENTED** [review]
Since this bug was not detected in the tests, IMHO, we need to update them to avoid reintroducing it in the future. Could you please add a new test or update an existing one to cover this call?

File: airflow/providers/databricks/operators/databricks.py (hunk 1 above)
Comment (potiuk) [reply airflow/providers/databricks/operators/databricks.py:524]: It's a mistery . MyPy uses a number of heuristics to find types of particular objects and sometime it gives up or sometimes drop types if they are too complex to analyse. Generally MyPy's behaveriour is often non-deterministic/difficult to reason about.

**Comment by adam133** [comment]
> Would you mind adding tests for the DatabricksHook to cover this for the future please?

I can try, but it'll take me a bit to understand how the testing code works. Might need some pointers

**Comment by potiuk** [comment]
> I can try, but it'll take me a bit to understand how the testing code works. Might need some pointers

Best to look at other, neighbouring tests and do it in s similar way. TESTING.rst contains detailed description of different kinds of tests we have and how to run them locally

**Comment by adam133** [comment]
> > I can try, but it'll take me a bit to understand how the testing code works. Might need some pointers
> 
> Best to look at other, neighbouring tests and do it in s similar way. TESTING.rst contains detailed description of different kinds of tests we have and how to run them locally

Alright, took a bit but I got some good tests in there. Found a few more bugs along the way.

**Review by potiuk — APPROVED** [review]

**Pull request merged**